### PR TITLE
My sql case insensitivity during Introspection

### PR DESCRIPTION
**Pull request by do4gr**
Fixes https://github.com/prisma/prisma/issues/6117

### CODE REVIEW

**Review by pimeys — APPROVED** [review]
jeez...

File: libs/sql-schema-describer/src/mysql.rs [hunk 1]
```diff
@@ -437,9 +437,9 @@ impl SqlSchemaDescriber {
             SELECT DISTINCT
                 index_name AS index_name,
                 non_unique AS non_unique,
-                column_name AS column_name,
+                Binary column_name AS column_name,
```
Comment (tomhoule) [inline libs/sql-schema-describer/src/mysql.rs:440]: I don't know that binary qualifier, is it something like a coercion?

File: introspection-engine/connectors/sql-introspection-connector/src/calculate_datamodel.rs [hunk 2]
```diff
@@ -33,8 +33,10 @@ pub fn calculate_datamodel(
     deduplicate_relation_field_names(&mut data_model);
 
     let mut warnings = vec![];
-    warnings.append(&mut enrich(previous_data_model, &mut data_model, family));
-    tracing::debug!("Enriching datamodel is done: {:?}", data_model);
+    if !previous_data_model.is_empty() {
+        warnings.append(&mut enrich(previous_data_model, &mut data_model, family));
+        tracing::debug!("Enriching datamodel is done: {:?}", data_model);
+    }
```
Comment (tomhoule) [inline introspection-engine/connectors/sql-introspection-connector/src/calculate_datamodel.rs:39]: is it related to the ordering change?

Comment (tomhoule) [reply introspection-engine/connectors/sql-introspection-connector/src/calculate_datamodel.rs:39]: nvm I just saw the commit message

File: libs/sql-schema-describer/src/mysql.rs (hunk 1 above)
Comment (do4gr) [reply libs/sql-schema-describer/src/mysql.rs:440]: That forces the comparisons to happen in a case sensitive way, without it Distinct collapses ADDRESSID, AddressId and addressId into one -.-

Comment (tomhoule) [reply libs/sql-schema-describer/src/mysql.rs:440]: ugh

**Pull request merged**